### PR TITLE
adjust offlineScoreAPI

### LINKED ISSUE #1014: LLSE 离线读写计分板API 报错问题与修改建议
### 异常模块

ScriptEngine (脚本引擎)

### 操作系统

Windows Server 2019

### LiteLoader 版本

2.9.1

### BDS 版本

1.19.51

### 发生了什么?

1、新建存档，没有玩家进过服，获取不存在的计分项（a）值报错。
2、新建存档，没有玩家进过服，设置玩家计分板值（a）报错，但自动了创建计分项（a）。
3、新建存档，玩家进服，计分项（a）已创建，获取玩家计分项（a）值报错。
4、新建存档，玩家进服，计分项（a）已创建，获取玩家【不存在的计分项（b）】值报错。
5、新建存档，玩家进服，计分项（a）已创建，获取【不存在的玩家】计分项（a）值报错。
以上问题是一连串的，详情见附录图片。
（以上问题在新建存档发生）
.
6、**离线获取计分项的api**，
**计分项a**已创建，但玩家没有设定该**计分项a**的值，但mc.getPlayerScore('uuid', 'a')返回值为**0**，这个应该返回**null**才对。
因为我在做跨服计分板同步，如果返回**0**会造成另一服务器同步计分板值为0，而实际该玩家没有该计分项的值。

### 复现此问题的步骤

_No response_

### 有关的日志/输出

_No response_

### 插件列表

_No response_

### PR DESCRIPTION
**Pull request by kkj9333**
## Description
Resolve https://github.com/LiteLDev/LiteLoaderBDS/issues/1014
Removed exception report
adjusted offlineScoreAPI

PR DESCRIPTION
#1014 
## Issues fixed

_None_

### CODE REVIEW

**Review by ShrBox — DISMISSED** [review]

File: LiteLoader/src/llapi/scoreboard/ScoreboardAPI.cpp [hunk 1]
```diff
@@ -60,14 +60,16 @@ LIAPI ScoreboardId Scoreboard::getOrCreateScoreboardId(mce::UUID const& uuid) {
 
 LIAPI bool Scoreboard::forceModifyPlayerScore(mce::UUID const& uuid, std::string const& objname, int val,
                                               PlayerScoreSetFunction pf) {
-    Objective* obj = Global<Scoreboard>->getObjective(objname);
-    if (!obj)
-        obj = Scoreboard::newObjective(objname, objname);
     auto Sid = Global<Scoreboard>->getOrCreateScoreboardId(uuid);
```
Comment (Jaciezyt) [inline LiteLoader/src/llapi/scoreboard/ScoreboardAPI.cpp]: Here the naming is not standardized.

**Review by Jaciezyt — CHANGES_REQUESTED** [review]

File: LiteLoader/src/llapi/scoreboard/ScoreboardAPI.cpp (hunk 1 above)
Comment (kkj9333) [reply LiteLoader/src/llapi/scoreboard/ScoreboardAPI.cpp]: Sid?

File: ScriptEngine/src/api/PlayerAPI.cpp [hunk 2]
```diff
@@ -332,8 +332,11 @@ Local<Value> McClass::getPlayerScore(const Arguments& args) {
     try {
         auto uuid = mce::UUID::fromString(args[0].asString().toString());
         auto obj = args[1].asString().toString();
-        auto score = Scoreboard::queryPlayerScore(uuid,obj).value();
-        return Number::newNumber(score);
+        auto scorev = Scoreboard::queryPlayerScore(uuid,obj);
+        if (scorev.has_value())
+            return Number::newNumber(scorev.value());
+        else
+            return Local<Value>();
```
Comment (Yushu2606) [inline ScriptEngine/src/api/PlayerAPI.cpp:339]: 应该是`return {}`？

Comment (kkj9333) [reply ScriptEngine/src/api/PlayerAPI.cpp:339]: 参考的其他返回空值时的返回值

**Review by ShrBox — APPROVED** [review]

File: LiteLoader/src/llapi/scoreboard/ScoreboardAPI.cpp (hunk 1 above)
Comment (Jaciezyt) [reply LiteLoader/src/llapi/scoreboard/ScoreboardAPI.cpp]: variable name should start with lower-case letter

Comment (kkj9333) [reply LiteLoader/src/llapi/scoreboard/ScoreboardAPI.cpp]: ok

**Review by dreamguxiang — APPROVED** [review]

**Pull request merged**